doc/Container Hardening/SONiC_container_hardening_HLD.md

```diff
@@ -0,0 +1,349 @@
+					**-p 6379:6379** \
+					-e RUNTIME_OWNER=local \
+					--uts=host \
+					--log-opt max-size=2M --log-opt max-file=5 \
+					--tmpfs /tmp \
+					$DB_OPT \
+					$REDIS_MNT \
+					-v /usr/share/sonic/device/$PLATFORM:/usr/share/sonic/platform:ro \
+					--tmpfs /var/tmp \
+					--env "NAMESPACE_ID"="$DEV" \
+					--env "NAMESPACE_PREFIX"="$NAMESPACE_PREFIX" \
+					--env "NAMESPACE_COUNT"=$NUM_ASIC \
+					--name=$DOCKERNAME \
+					docker-database:latest \
+
+**How we did it?**
+
+To create a docker with the flags above it is required to set the "new" flag in the file docker_image_ctl.js. Follow the call `docker create {{docker_image_run_opt}} \`: 
+and replace the `–--net=$NET`.
+docker flag generation
+
+	{%- if docker_container_name != "database" %}
+					--net=$NET \
+	{%- endif %}
+	{%- if docker_container_name == "database" %}
+					-p 6379:6379 \
+	{%- endif %}
+
+#### How to check?
+
+Go into the docker - `docker exec -it docker bash`
+Run `ifconfig`.
+
+On a docker with host network - you'll be able to view all physical interfaces.
+On a docker without host network - we'll see only eth0 and lo.
+
+## 5. SAI API
+
+N/A
+
+## 6. Configuration and management
+
+N/A - no configuration management/changes are required.
+
+### 6.1. Manifest (if the feature is an Application Extension)
+
+N/A
+
+### 6.2. CLI/YANG model Enhancements
+
+N/A
+We are not adding CLI commands or management capabilities to the system with this item.
+
+### 6.3. Config DB Enhancements
+
+N/A - DB should remain the same
+
+## 7. Warmboot and Fastboot Design Impact
+
+No impact on all boot sequences, as this item should be seemlessly integrated into the system and achieve the same functionality level as before.
+
+## 8. Restrictions/Limitations
+
+## 9. Testing Requirements/Design
+
+To define this item completed - we are required to run the full CI and check that nothing has been broken from the changes proposed in this HLD.
+In addition - we should test that the mitigations are applicable for the relevant containers.
+
+### 9.1 Unit Test cases
+
+N/A, this feature will be checked on a system level.
+
+### 9.2 System Test cases
+
+For general fucntionality flows-  running the same test cases that we currently have on top of our system and verifying that nothing broke.
+
+For adidtional security test cases, we should check that priviliges and network capabilities have been removed.
+Net=$HOST removal test:
+1. Login to container with removed network capabilities
+2. Run ls /dev/
+3. Check that we do not have visibility to all network devices (no tty9/8 no sda, etc')
+
+Privilege removal test:
+1. Login to container without --privileged flag
+2. Check that you cannot access /etc/shadow
+3. Check that you cannot perform vim for /boot folder or any file in it
+
+## 10. Open/Action items - if any
+
+Currently, Nvidia and MSFT have scoped commitment for specific containers.
+Redis and SNMP already have these adjustments.
```

Comment:
What if SNMP container wants to listen some particular interface on the host? container hardening will be disabled by default? can this be driven based on FEATURE table config?

Reply:
If the SNMP container would like to listen to a specific interface - you can check this:
https://github.com/sonic-net/sonic-buildimage/pull/15176/files

I think that this PR will showcase your specific use-case.